File: ts-playwright-cucumber/src/pages/checkoutCompletePage.ts
import type { Page } from 'playwright';
import { BasePage } from './basePage.ts';

const COMPLETE_HEADER = "[data-test='complete-header']";

export class CheckoutCompletePage extends BasePage {
  constructor(page: Page) {
    super(page);
  }

  getConfirmationMessage(): Promise<string> {
    return this.textOf(COMPLETE_HEADER);
  }
}

File: csharp-playwright-nunit-reqnroll/SauceDemo.UiTests/Pages/CheckoutCompletePage.cs
using Microsoft.Playwright;

namespace SauceDemo.UiTests.Pages;

public sealed class CheckoutCompletePage : BasePage
{
    private const string CompleteHeader = "[data-test='complete-header']";

    public CheckoutCompletePage(IPage page) : base(page)
    {
    }

    public Task<string> GetConfirmationMessageAsync() => TextOfAsync(CompleteHeader);
}
